src/views/Components/PlayerSettings.tsx

```diff
@@ -0,0 +1,104 @@
+import { useState } from 'react';
+import Button from '../../elements/Button'
+import arrowButton from '../../assets/svg-icons/arrow.svg';
+
+function createPlayers() {
+    return [
+        {
+            id: 0,
+            name: "Player 1",
+        },
+        {
+            id: 1,
+            name: "Player 2",
+        },
+        {
+            id: 2,
+            name: "Player 3",
+        },
+        {
+            id: 3,
+            name: "Player 4",
+        },
+    ];
+}
+
+function Settings() {
+
+    const [players, setPlayers] = useState(createPlayers());
+
+    const newId = Math.max(...players.map(p => p.id)) + 1
+
+    const isThereLessThanTwelvePlayers = players.length < 12;
+    const isThereMoreThanFourPlayers = players.length > 4;
+    const isThisTheFirstPlayer = (i: number) => i === players[0].id ? true : false;
+    const isThisTheLastPlayer = (i: number) => i === players[players.length - 1].id;
+
+    function handlePlayerNameChange(e: React.ChangeEvent<HTMLInputElement>, i: number) {
+        const copyPlayers = [...players];
+        copyPlayers[i].name = e.target.value;
+        setPlayers(copyPlayers);
+
+    };
+
+    function addPlayerNameClick() {
+        const copyPlayers = [...players];
+        //player.name needs to be 1 value higher than player.id because the index starts at 0 and the name starts at "Player 1"
+        copyPlayers.push({ id: newId, name: `Player ${newId + 1} ` });
```

Comment:
It would be better to put the id creation logic you have above in this function here. So it would be something like:
```
const copyPlayers = [...players];
const id = Math.max(...players.map(p => p.id)) + 1;
copyPlayers.push({ id, name: `Player ${id} ` });
```